aiodocker/docker.py

```diff
@@ -113,11 +113,24 @@ def __init__(self,
         data = await self._query_json("version")
         return data
 
-    async def pull(self, image, *, stream=False):
+    # maybe discard future
+    async def pull(self, image, auth=None, stream=False):
+
+        headers = {"content-type": "application/json"}
+        if auth:
+            if isinstance(auth, dict) and 'auth' in auth:
+                registry, p, _ = image.partition('/')
+                if not p:
+                    raise ValueError(" image should have registry host")
+                auth_header = parse_base64_auth(auth['auth'], registry)
+                headers.update({"X-Registry-Auth": auth_header})
+            else:
+                raise ValueError(" auth format error " + str(auth))
```

Comment:
I have add 2 ValueError test in test_images --> test_pups_image_auth with pytest.raise(Exception)